server/protocols/webadmin/README.adoc

```diff
@@ -0,0 +1,121 @@
+= Web administration for JAMES
+
+The web administration supports for now the CRUD operations on the domains and the users, as described in the following sections.
+
+**WARNING** : This API do not allow authentication for now. It means that an administrator should ensure an attacker can not use this API.
+
+== Administrating domains
+
+=== Create a domain
+
+.bash
+====
+curl -XPUT http://ip:port/domains/domainToBeCreated
+====
+
+Resource name domainToBeCreated :
+ - can not be null or empty
+ - can not contain '@'
+ - can not be more than 255 characters
+ - can not contain '/'
+
+Response codes :
+ - 204 : The domain was successfully added
+ - 500 : Internal error while adding the domain
+
+=== Delete a domain
+
+.bash
+====
+curl -XDELETE http://ip:port/domains/domainToBeDeleted
+====
+
+Response codes :
+ - 204 : The domain was successfully removed
+ - 500 : Internal error while removing the domain
+
+=== Test if a domain exists
+
+.bash
+====
+curl -XGET http://ip:port/domains/domainName
+====
+
+Response codes :
+ - 204 : The domains exists
+ - 404 : The domain do not exist
+ - 500 : Internal error while accessing the domains
+
+=== Get the list of domains
+
+.bash
+====
+curl -XGET http://ip:port/domains
+====
+
+Possible response :
+
+.json
+====
+{"domains":["domain1", "domain2"]}
+====
+
+Response codes :
+ - 200 : The domain list was successfully retrieved
+ - 500 : Internal error while accessing the domains
+
+== Administrating users
+
+=== Create a user
+
+.bash
+====
+curl -XPUT http://ip:port/users/usernameToBeUsed -d '{"password":"passwordToBeUsed"}'
+====
+
+Resource name usernameToBeUsed :
+ - can not be null or empty
+ - can not be more than 255 characters
+ - can not contain '/'
+
+Response codes :
+ - 204 : The user was successfully created
+ - 409 : Conflict : the user was removed between existence check and insert / update
```

Comment:
Yes, because the user also have a password. If we are victim of a data race, then previous password is kept. Hence our operation is not indepotant and we need to report the error.